src/scheduler.jl

```diff
@@ -57,9 +57,41 @@ function compute_dag(ctx, d::Thunk)
             continue
         end
 
+        function check_exited_exception(res::CapturedException)
+            check_exited_exception(res.ex)
+        end
+        function check_exited_exception(res::RemoteException)
+            check_exited_exception(res.captured)
+        end
+        check_exited_exception(res::ProcessExitedException) = true
+        check_exited_exception(res) = false
+
         proc, thunk_id, res = take!(chan)
         if isa(res, CapturedException) || isa(res, RemoteException)
-            throw(res)
+            # TODO: showerror(res)
+            if check_exited_exception(res)
+                @warn "Worker $(proc.pid) died, rescheduling work"
+                #=
+                for tid in keys(state.thunk_dict)
+                    inps = state.thunk_dict[tid].inputs
+                    @info "TID $tid: $inps"
+                end
+                =#
+                # TODO: Schedule more intelligently
+                newproc = OSProc(rand(workers()))
+                fire_task!(ctx, state.thunk_dict[thunk_id], newproc, state, chan, node_order)
+                continue
```

Comment:
This is a great start, it should work for when all the inputs are coming from disk or as non-chunk arguments. But won't work if the required input was on the worker that failed...

Reply:
Thanks! To clarify for my benefit, you mean that this isn't currently handling the case where the result of an input thunk only exists on the dead worker, so we have to go back and recompute all "missing" inputs?

Reply:
@shashi any chance I can get a hint on how to figure out where a completed dependent thunk resides (on which worker process)? :slightly_smiling_face:

Reply:
Nevermind I figured it out. Now I need to get all the state just right so it stops throwing `KeyError`s at me...